includes/DefaultSettings.php

```diff
@@ -3373,6 +3490,150 @@
+// from various log pages by default
+$wgGroupPermissions['bot']['bot']              = true;
+$wgGroupPermissions['bot']['autoconfirmed']    = true;
+$wgGroupPermissions['bot']['nominornewtalk']   = true;
+$wgGroupPermissions['bot']['autopatrol']       = true;
+$wgGroupPermissions['bot']['suppressredirect'] = true;
+$wgGroupPermissions['bot']['apihighlimits']    = true;
+$wgGroupPermissions['bot']['writeapi']         = true;
+#$wgGroupPermissions['bot']['editprotected']    = true; // can edit all protected pages without cascade protection enabled
+
+// Most extra permission abilities go to this group
+$wgGroupPermissions['sysop']['block']            = true;
+$wgGroupPermissions['sysop']['createaccount']    = true;
+$wgGroupPermissions['sysop']['delete']           = true;
+$wgGroupPermissions['sysop']['bigdelete']        = true; // can be separately configured for pages with > $wgDeleteRevisionsLimit revs
+$wgGroupPermissions['sysop']['deletedhistory']   = true; // can view deleted history entries, but not see or restore the text
+$wgGroupPermissions['sysop']['deletedtext']      = true; // can view deleted revision text
+$wgGroupPermissions['sysop']['undelete']         = true;
+$wgGroupPermissions['sysop']['editinterface']    = true;
+$wgGroupPermissions['sysop']['editusercss']      = true;
+$wgGroupPermissions['sysop']['edituserjs']       = true;
+$wgGroupPermissions['sysop']['import']           = true;
+$wgGroupPermissions['sysop']['importupload']     = true;
+$wgGroupPermissions['sysop']['move']             = true;
+$wgGroupPermissions['sysop']['move-subpages']    = true;
+$wgGroupPermissions['sysop']['move-rootuserpages'] = true;
+$wgGroupPermissions['sysop']['patrol']           = true;
+$wgGroupPermissions['sysop']['autopatrol']       = true;
+$wgGroupPermissions['sysop']['protect']          = true;
+$wgGroupPermissions['sysop']['proxyunbannable']  = true;
+$wgGroupPermissions['sysop']['rollback']         = true;
+$wgGroupPermissions['sysop']['upload']           = true;
+$wgGroupPermissions['sysop']['reupload']         = true;
+$wgGroupPermissions['sysop']['reupload-shared']  = true;
+$wgGroupPermissions['sysop']['unwatchedpages']   = true;
+$wgGroupPermissions['sysop']['autoconfirmed']    = true;
+$wgGroupPermissions['sysop']['upload_by_url']    = true;
+$wgGroupPermissions['sysop']['ipblock-exempt']   = true;
+$wgGroupPermissions['sysop']['blockemail']       = true;
+$wgGroupPermissions['sysop']['markbotedits']     = true;
+$wgGroupPermissions['sysop']['apihighlimits']    = true;
+$wgGroupPermissions['sysop']['browsearchive']    = true;
+$wgGroupPermissions['sysop']['noratelimit']      = true;
+$wgGroupPermissions['sysop']['movefile']         = true;
+$wgGroupPermissions['sysop']['unblockself']      = true;
+$wgGroupPermissions['sysop']['suppressredirect'] = true;
+#$wgGroupPermissions['sysop']['mergehistory']     = true;
+
+// Permission to change users' group assignments
+$wgGroupPermissions['bureaucrat']['userrights']  = true;
```

Comment:
Do this get overridden anywhere? We don't want bureaucrats to have the `userrights` permission as it allows changing all groups, not just the ones we want to limit them to (we don't even give `userrights` to the `util` group).

I'd also want to check these match up and don't give extra permissions than those defined in `PermissionsDefinesBeforeWikiFactory.php` in config. I know this is resetting the file to the default MW 1.19, but since all the permissions config was pulled out to `PermissionsDefines*WikiFactory.php, I'd consider just removing all these from the file again, especially since the plan is for all the permission config to be pulled out of MW and into the user-permissions service. What do you think?

Reply:
@Grunny - it is discarded completely in `includes/wikia/DefaultSettings.php`: https://github.com/Wikia/app/pull/14597/files#diff-013f1f772182745f48c5f1429d0cdd35R99 - all FANDOM-specific overrides to `includes/DefaultSettings.php` are in `includes/wikia/DefaultSettings.php`, the section starts here: https://github.com/Wikia/app/pull/14597/files#diff-013f1f772182745f48c5f1429d0cdd35R23

I am hesitant about chaning `includes/DefaultSettings.php`.  Keeping it vanilla and having all overrides explicitly stated can help to understand what we changed and why.  We also randomly port pieces of code from newer versions of MediaWiki, so the ability to `diff` our file with the one from any newer version of MediaWiki might be helpful.

`DefaultSettings.php` clearly states _do not edit_, so changing it might be confusing to our younger developers, as they would expect it to be vanilla MediaWiki, untouched by us.

On the other hand, we may make a slight change in the core logic and remove `includes/DefaultSettings.php` completely in favour of `includes/wikia/DefaultSettings.php` and set all required defaults there.  It would address your remarks, as well as force developers who import fresh MW stuff (essentially @TK-999 , @macbre and myself) to consult the original `DefaultSettings.php` and port certain defaults to our `includes/wikia/DefaultSettings.php`.

Also, in the upcoming stages, I will be moving stuff from `CommonExtensions.php` to Wikia/app - as it not configuration, just importing libraries (after some cleanup).  In addition I will be merging `CommonSettings.php` and `config/LocalSettings.php` into some base config, on top of which we'll have environment-specific overrides as well as `PermissionDefines*WikiFactory.php`, and WikiFactory injected in the right place: all that in `app/LocalSettings.php`. There will be some overrides after WikiFactory, although we should aim for WikiFactory being the final layer.